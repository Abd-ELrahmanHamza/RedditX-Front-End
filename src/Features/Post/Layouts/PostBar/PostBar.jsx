import PostPublisher from "Features/Post/Components/PostPublisher/PostPublisher";
import PostContent from "Features/Post/Components/PostContent/PostContent";
import PostFooter from "Features/Post/Components/PostFooter/PostFooter";

import {Container} from "./PostBar.styled"

/**
 * PostPublisher Component 
 * @returns {React.Component} PostPublisher component
 */
const PostBar = ({ fullPost, post, setMakeHidden }) => {
  return (
    <Container>
      <PostPublisher fullPost={fullPost} post={post} />
      <PostContent post={post} />
      <PostFooter post={post} setMakeHidden={setMakeHidden}/>
    </Container>
  );
};

export default PostBar;
