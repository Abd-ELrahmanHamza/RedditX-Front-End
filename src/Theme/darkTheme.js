/**
 * dark theme colors
 *
 * @type {{id:string}}
 */
const darkTheme = {
  // The id of the theme
  id: "dark",
  // The canvas color is the background color of the website
  canvas: "#030303",
  // The colors of the text on the website
  color: {
    // The primary color of the text on the website
    primary: "#d7dadc",
    // The color of the muted text on the website
    muted: "#7c7c7c",
    // The color of the links on the website
    link: "#0079d3",
  },
  // The colors of the backgrounds in the website
  background: {
    primary: "#1a1a1b",
  },
  // The colors of the lines in the website
  lineColor: {
    primary: "#343536",
  },
};

export default darkTheme;
