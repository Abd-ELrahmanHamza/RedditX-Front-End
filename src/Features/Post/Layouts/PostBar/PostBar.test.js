import { shallow } from "enzyme";
import { screen, render, waitFor } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import darkTheme from "Theme/darkTheme";
import PostBar from "./PostBar";
import { BrowserRouter } from "react-router-dom";

describe("PostBar Component", () => {
  const wrapper = shallow(<PostBar />);

  it("this is a test for PostBar Component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
