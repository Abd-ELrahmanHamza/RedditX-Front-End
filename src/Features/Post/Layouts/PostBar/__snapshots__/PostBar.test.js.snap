// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostBar Component this is a test for PostBar Component 1`] = `
<styled.div>
  <PostPublisher
    fullPost={true}
  />
  <PostContent />
  <PostFooter />
</styled.div>
`;
